src/endpoints/authentication/impl.cpp

```diff
@@ -254,7 +257,30 @@ namespace irods::http::handler
 						return _sess_ptr->send(fail(status_type::bad_request));
 					}
 
-					auto is_state_valid{[](std::string_view in_state) { return in_state.compare("placeholder") == 0; }};
+					auto is_state_valid{[](const std::string& _in_state) {
+					  // Search to see if state exists
+					  auto mapped_value{irods::http::process_stash::find(_in_state)};
+					  if (!mapped_value) {
+						return false;
+					  }
+					  auto* expire_time{boost::any_cast<std::chrono::steady_clock::time_point>(&*mapped_value)};
+
+					  // It exists!!! Let's see what we can do...
+					  if (expire_time != nullptr) {
+						// Ensure state is not expired...
+						const auto now{std::chrono::steady_clock::now()};
+						const auto is_not_expired{now < *expire_time};
+
+						// Remove item from valid states
+						irods::http::process_stash::erase(_in_state);
```

Comment:
We should check the return value of this, in the case two requests use the same state value. Only one should be valid.

Reply:
Done